q_yaar_platform/treasure_hunt/models.py

```diff
@@ -0,0 +1,66 @@
+import uuid
+import pghistory
+
+from django.conf import settings
+from django.db import models
+
+from common.abstract_models import AbstractExternalFacing, AbstractGame, AbstractTimeStamped, AbstractVersioned
+from treasure_hunt.popo.time_interval import TimeIntervalListConfig
+
+
+@pghistory.track()
+class TreasureHuntGame(AbstractGame, AbstractExternalFacing, AbstractTimeStamped, AbstractVersioned):
+    CONST_KEY_GAME_DURATION = "game_duration"
+    CONST_KEY_START_TIMESTAMP = "start_timestamp"  # epoch timestamp
+    CONST_KEY_END_TIMESTAMP = "end_timestamp"  # epoch timestamp
+
+    info = models.JSONField(default=dict)
+
+    class Meta:
+        indexes = [models.Index(fields=["game_code"])]
+
+    def __str__(self) -> str:
+        return f"{self.game_code}"
+
+    def get_game_duration(self) -> TimeIntervalListConfig:
```

Comment:
This will be a list of epoch timestamp intervals. During the game we will find and cache the current interval. Once the timestamp is past the current interval no activities will be allowed.
For enhancing we will also cache the next interval so that next day start isn't affected. Implementation can be discussed later.

More on the structure in the below popo.